### PR TITLE
Don't compute metrics in LM examples on TPU

### PR DESCRIPTION
**Pull request by sgugger**
# What does this PR do?

This PR removes the accuracy computation in the LM examples on TPU, as TPUs can't handle the accumulation of the model logits.

Patches #16005 A better fix would be to be able to gather those logits on TPU but waiting for advice from the pytorch XLA team on this.

### CODE REVIEW

**Comment by miladm** [comment]
Thank you for the fix!

**Review by LysandreJik — APPROVED** [review]
LGTM
